tests/framework/results/test_manager.py

```diff
@@ -397,8 +400,62 @@ def test_unused_stratifications_are_logged(caplog):
     assert "['familiar']" in log_split[1]
 
 
-def test_update_monotonically_increasing__raw_results():
-    """Test that (monotonically increasing) raw results are being updated correctly."""
+def test_stratified_observation_results():
+    components = [
+        Hogwarts(),
+        CatLivesObserver(),
+        HogwartsResultsStratifier(),
+    ]
+    sim = InteractiveContext(configuration=HARRY_POTTER_CONFIG, components=components)
+    assert (sim.get_results()["cat_lives"]["value"] == 0.0).all()
+    sim.step()
+    num_familiars = sim.get_population().groupby(["familiar", "student_house"]).apply(len)
+    expected = num_familiars.loc["cat"] * 9.0
+    expected.name = "value"
+    assert expected.sort_values().equals(
+        sim.get_results()["cat_lives"]["value"].sort_values()
+    )
+
+
+def test_unstratified_observation_results():
+    components = [
+        Hogwarts(),
+        ValedictorianObserver(),
+    ]
+    sim = InteractiveContext(configuration=HARRY_POTTER_CONFIG, components=components)
+    sim.step()
+    first_valedictorian = sim.get_results()["valedictorian"]
+    assert len(first_valedictorian) == 1
+    sim.step()
+    second_valedictorian = sim.get_results()["valedictorian"]
+    assert len(second_valedictorian) == 1
+    assert (
+        first_valedictorian["student_id"].iat[0] != second_valedictorian["student_id"].iat[0]
```

Comment:
Technically this could fail if the RNG gods really hate us and we choose the same simulant both times?